config/clusters/2i2c/neurohackademy.values.yaml

```diff
@@ -56,6 +57,12 @@ jupyterhub:
           http://github.com/login/oauth/authorize:
             username_derivation:
               username_claim: "preferred_username"
+    extraFiles:
+      configurator-schema-default:
```

Comment:
```suggestion
      # We override the configuration of jupyterhub-configurator as we have configured singleuser.defaultUrl
      # above, and no longer wants the user to override this
      configurator-schema-default:
```

Reply:
```suggestion
      # We disable the defaultUrl set by the jupyterhub-configurator schema,
      # otherwise it would override the singleuser.defaultUrl set above.
      # This is because the jupyterhub-configurator schema gets loaded after the hub
      # yaml config and overrides any config set before.
      # Ref: https://infrastructure.2i2c.org/en/latest/topic/config.html#inheritance-of-configuration
      configurator-schema-default:
```

Reply:
Actually, setting `defaultUrl` to null in configurator's schema doesn't disable the configurator for changing the interface. You can still do that.

It just doesn't set a default again(https://github.com/2i2c-org/infrastructure/blob/a3cf596562e374aa01df8cc70d5a26b4817111a7/helm-charts/basehub/values.yaml#L189) overriding whatever we set in the hub's config file.

I added a suggestion above with what the explanation I think it should be. It also has a link to our docs where there a list with the order in which each config file gets loaded and can override each other. Do you think that is enough?

Reply:
Thanks for clarifying how things work! This is absolutely adequate, the key part in my suggestions were to somehow link together the configuration so that its not two separate parts that have a clear connection - but that connection isn't seen by readers of the config.

Reply:
Makes sense! Thanks @consideRatio ✨